classes/01class/exercises/c01-git03/gpavelar/README.md

```diff
@@ -0,0 +1,53 @@
+# c01-git03
+
+## Questions (answer inline)
+
+1. Did you notice any difference in the size of the repo before and after adding the big file?
+
+> According to those commands `du -sh` and `du -h` the size of the repo was around 4.0K before `git init` command.
+> After that the repository created locally and run all commands required it was around 170K. The command `git init` creates hidden files under the directory `.git` containing all metadate for this new repository to keep track of everything.
+> Adding a big file into this repository its size increase a bit bit, size around 125M (I created another file a little bit bigger). When you add this file into the repository, .git directory increases its size due to the fact it now trying to store/manage this big file that its size is above git limits. Eventually, GitHub will block pushes that contain files over 100MB.
+
+2. What is the reason for this problem?
+
+> Git was created originally to handle source code that was quite small. Github has a size limit for files to not overload servers and decrease the performance. And it keeps track of any change in a binary file (whole history) as well.
+> So it could be a serious problem when cloning repositories and pushing them.
+> A simple way to overcome it using `git lfs` (git large filesystem) 'extension' on the computer. It creates a pointer to your 'big file' that is stored into git lfs system.
+
+3. How do you remove something from Git history after it is pushed to the remote repository? Which commands would you use?
+
+> You are able to remove anything from git history, but it depends on what you want. On this case with the big file from previous exercise.
+
+```
+git rm --cached file.txt
+git commit --amend -CHEAD --allow-empty
+git push --force --all
```

Comment:
the assumption here is that you're fixing the last commit. Otherwise, that would be a `rebase -i`

Reply:
Sorry, 
> How do you remove something from Git history after it is pushed to the remote repository? Which commands would you use?

I understood it wrongly. If you run git rebase --interactive on something that was pushed, you get the same pointer to same master (that you've pushed), right? It won't remove something from git history.
my Notes: rebase is just to keep a linear project history, isn't it ?

Should I change my answer?

Reply:
Your answer is correct too. It is considering that only the last commit will need to change, keeping the rest of the history as it is. 

The meaning of `remove` in the question is about removing commits, not files (my bad).

In `git rebase -i` you have some options to apply to commits, as below.
So you can use `drop` for example for removing a commit anywhere in the history.

```
# Commands:
# p, pick <commit> = use commit
# r, reword <commit> = use commit, but edit the commit message
# e, edit <commit> = use commit, but stop for amending
# s, squash <commit> = use commit, but meld into previous commit
# f, fixup <commit> = like "squash", but discard this commit's log message
# x, exec <command> = run command (the rest of the line) using shell
# b, break = stop here (continue rebase later with 'git rebase --continue')
# d, drop <commit> = remove commit
# l, label <label> = label current HEAD with a name
# t, reset <label> = reset HEAD to a label
# m, merge [-C <commit> | -c <commit>] <label> [# <oneline>]
# .       create a merge commit using the original merge commit's
# .       message (or the oneline, if no original merge commit was
# .       specified). Use -c <commit> to reword the commit message.
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
```